internal/services/hybridcompute/arc_machine_extension_resource_test.go

```diff
@@ -0,0 +1,183 @@
+}
+
+func TestAccHybridComputeMachineExtension_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_arc_machine_extension", "test")
+	r := HybridComputeMachineExtensionResource{}
+	template := r.template(data)
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data, template),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("protected_settings"),
+		{
+			Config: r.update(data, template),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("protected_settings"),
+	})
+}
+
+func (r HybridComputeMachineExtensionResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := machineextensions.ParseExtensionID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	client := clients.HybridCompute.MachineExtensionsClient
+	resp, err := client.Get(ctx, *id)
+	exists := false
+	if err != nil {
+		if response.WasNotFound(resp.HttpResponse) {
+			return &exists, nil
+		}
+		return nil, fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+	return pointer.To(resp.Model != nil), nil
+}
+
+func (r HybridComputeMachineExtensionResource) basic(data acceptance.TestData, template string) string {
+	return fmt.Sprintf(`
+				%s
+
+resource "azurerm_arc_machine_extension" "test" {
+  name                      = "acctest-hcme-%d"
+  arc_machine_id            = data.azurerm_arc_machine.test.id
+  publisher                 = "Microsoft.Azure.Monitor"
+  type                      = "AzureMonitorLinuxAgent"
+  location                  = "%s"
+}
+`, template, data.RandomInteger, data.Locations.Primary)
+}
+
+func (r HybridComputeMachineExtensionResource) requiresImport(basicConfig string) string {
+	return fmt.Sprintf(`
+			%s
+
+resource "azurerm_arc_machine_extension" "import" {
+  name                      = azurerm_arc_machine_extension.test.name
+  arc_machine_id            = azurerm_arc_machine_extension.test.arc_machine_id
+  publisher                 = azurerm_arc_machine_extension.test.publisher
+  type                      = azurerm_arc_machine_extension.test.type
+  location                  = azurerm_arc_machine_extension.test.location
+}
+`, basicConfig)
+}
+
+func (r HybridComputeMachineExtensionResource) complete(data acceptance.TestData, template string) string {
+	return fmt.Sprintf(`
+			%s
+
+resource "azurerm_arc_machine_extension" "test" {
+  name                               = "acctest-hcme-%d"
+  arc_machine_id                     = data.azurerm_arc_machine.test.id
+  location                           = "%s"
+  automatic_upgrade_enabled          = false
+  publisher                          = "Microsoft.Azure.Monitor"
+  type                               = "AzureMonitorLinuxAgent"
+  type_handler_version               = "1.24"
+}
+`, template, data.RandomInteger, data.Locations.Primary)
+}
```

Comment:
It looks like this config is missing a fair few properties for it to be complete?

Reply:
Let me add settings and protected settings example to complete config.